.github/workflows/build-linux.yml

```diff
@@ -0,0 +1,92 @@
+name: Linux tests
+
+on:
+  # This avoids having duplicate builds for a pull request
+  push:
+    branches:
+    - master
+    - 5.x
+    - 4.x
+  pull_request:
+    branches:
+    - master
+    - 5.x
+    - 4.x
+
+jobs:
+  cleanup-previous-runs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: rokroskar/workflow-run-cleanup-action@master
+      env:
+        GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+  build:
+    name: Py${{ matrix.PYTHON_VERSION }}, ${{ matrix.INSTALL_TYPE }}, ${{ matrix.TEST_TYPE }}
+    runs-on: ubuntu-latest
+    env:
+      CI: 'true'
+      CODECOV_TOKEN: "56731c25-9b1f-4340-8b58-35739bfbc52d"
+      OS: 'linux'
+      PYTHON_VERSION: ${{ matrix.PYTHON_VERSION }}
+      RUN_SLOW: ${{ matrix.TEST_TYPE == 'slow' }}
+      USE_CONDA: ${{ matrix.INSTALL_TYPE == 'conda' }}
+    strategy:
+      fail-fast: false 
+      matrix:
+        INSTALL_TYPE: ['conda', 'pip']
+        PYTHON_VERSION: ['3.7', '2.7']
+        TEST_TYPE: ['fast', 'slow']
+        exclude:
+          - INSTALL_TYPE: 'pip'
+            PYTHON_VERSION: '2.7'
+            TEST_TYPE: 'fast'
+          - INSTALL_TYPE: 'pip'
+            PYTHON_VERSION: '2.7'
+            TEST_TYPE: 'slow'
+    steps:
+      - name: Install dependencies
+        shell: bash
+        run: |
+          sudo apt-get install libxcb-randr0-dev libxcb-xtest0-dev libxcb-xinerama0-dev libxcb-shape0-dev libxcb-xkb-dev libegl1-mesa libxkbcommon-x11-0
+      - name: Checkout branch
+        uses: actions/checkout@v1.2.0
+      - name: Cache pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements/*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - name: Install Conda
+        uses: goanpeca/setup-miniconda@v1
+        with:
+           activate-environment: test
+           auto-update-conda: true
+           auto-activate-base: false
+           python-version: ${{ matrix.PYTHON_VERSION }} 
+      - name: Create test environment
+        shell: bash -l {0}
+        run: |
+          bash -l .github/workflows/install.sh
+      - name: Show test environment
+        shell: bash -l {0}
+        run: |
+          conda info
+          conda list
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          xvfb-run --auto-servernum python runtests.py
+      - name: Run application test
+        shell: bash -l {0}
+        run: |
+          xvfb-run --auto-servernum python bootstrap.py -- --reset
```

Comment:
This is really not  necessary unless the tests are run several times, so you could remove it.

Reply:
Nop, I mean `python bootstrap.py -- --reset`. The thing is if the test suite failed due to a segfault, we needed to reset the settings from the previous run so they don't interfere with the tests that depend on them in the new run.

Since you're running the tests only one time here, this is not necessary anymore.